contracts/dapis/DapiServer.sol

```diff
@@ -495,6 +495,40 @@ contract DapiServer is
         emit UpdatedBeaconWithSignedData(beaconId, decodedData, timestamp);
     }
 
+    /// @notice Updates a Beacon using data signed for this contract by the
+    /// respective Airnode, without requiring a request or subscription
+    /// @param airnode Airnode address
+    /// @param templateId Template ID
+    /// @param timestamp Timestamp used in the signature
+    /// @param data Response data (a `uint256` encoded in contract ABI)
+    /// @param signature Template ID, a timestamp and the response data signed
+    /// for this contract by the Airnode address
+    function updateBeaconWithDomainSignedData(
+        address airnode,
+        bytes32 templateId,
+        uint256 timestamp,
+        bytes calldata data,
+        bytes calldata signature
+    ) external override onlyValidTimestamp(timestamp) {
+        require(
+            (
+                keccak256(
+                    abi.encodePacked(
+                        block.chainid,
+                        address(this),
```

Comment:
I don't think we should have the contract address be added in the signature, in case we want to upgrade `dapiServer`, it would then require a redeployment across all providers. What is the benefit of having it be appended to the signature?

Reply:
The gateway for this will receive `block.chainid` and `address(this)` as API call parameters and sign accordingly, so it won't require a redeployment. You want the contract address in the signature so that if there is another `DapiServer.sol` (or another contract that uses signatures with the same format) is deployed on the same chain, the signatures for the two contracts can't be replayed on each other.

Reply:
I didn't know that the gateway received the chainId and address(this), looks good then 👍 . I thought about the signature replay but didn't think it was that much of an issue.